### PR TITLE
Update to point to latest version of gson

### PR DESCRIPTION
**Pull request by neha-bhargava**
Fixes # 
#3098 

**Changes proposed in this request**
Update pom.xml for cache compat tests for java to pull latest version for gson.

**Testing**

**Performance impact**

### CODE REVIEW

File: tests/CacheCompat/CommonCache.Test.MsalJava/pom.xml [hunk 1]
```diff
@@ -27,7 +27,7 @@
         <dependency>
             <groupId>com.google.code.gson</groupId>
             <artifactId>gson</artifactId>
-            <version>2.8.6</version>
+            <version>[2.8.6,)</version>
```
Comment (neha-bhargava) [inline tests/CacheCompat/CommonCache.Test.MsalJava/pom.xml:30]: Including `)` at the end will pick the latest version for this lib.

File: tests/CacheCompat/CommonCache.Test.MsalJava/pom.xml [hunk 2]
```diff
@@ -27,7 +27,7 @@
         <dependency>
             <groupId>com.google.code.gson</groupId>
             <artifactId>gson</artifactId>
-            <version>2.8.6</version>
+            <version>[2.8.9,)</version>
```
Comment (SameerK-MSFT) [inline tests/CacheCompat/CommonCache.Test.MsalJava/pom.xml:30]: There are characters [ at the beginning and ,) at the end. Typos?

**Review by SameerK-MSFT — CHANGES_REQUESTED** [review]

File: tests/CacheCompat/CommonCache.Test.MsalJava/pom.xml (hunk 2 above)
Comment (bgavrilMS) [reply tests/CacheCompat/CommonCache.Test.MsalJava/pom.xml:30]: No, this is a floating version. It means `any version >= 2.8.9`. Btw, nuget introduced them as well https://docs.microsoft.com/en-us/nuget/concepts/dependency-resolution#floating-versions

**Review by bgavrilMS — APPROVED** [review]
LGTM

**Review by pmaytak — APPROVED** [review]

File: tests/CacheCompat/CommonCache.Test.MsalJava/pom.xml (hunk 2 above)
Comment (neha-bhargava) [reply tests/CacheCompat/CommonCache.Test.MsalJava/pom.xml:30]: These are not typos. This means the min version is 2.8.9 and max is the latest. See https://docs.oracle.com/middleware/1212/core/MAVEN/maven_version.htm#MAVEN402

**Pull request merged**